pymc_marketing/metrics.py

```diff
@@ -0,0 +1,93 @@
+#   Copyright 2024 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Evaluation Metrics."""
+
+import numpy as np
+import numpy.typing as npt
+
+
+def per_observation_crps(y_true: npt.NDArray, y_pred: npt.NDArray) -> npt.NDArray:
```

Comment:
Is this per_observation or per_sample? Doesn't this get a distribution of the metric?

Maybe `calculate_crps_distribution` or something to hint that this is a distribution of a metric

Reply:
added an example in https://github.com/pymc-labs/pymc-marketing/pull/971/commits/d70e77ec88d5b85306fc7d9d419644c8cad55b1a